File: lib/src/config/models/source_location.dart
/// A source location within a configuration file.
final class SourceLocation {
  final String filePath;
  final int line;
  final int column;

  const SourceLocation({
    required this.filePath,
    required this.line,
    required this.column,
  });

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is SourceLocation &&
          filePath == other.filePath &&
          line == other.line &&
          column == other.column;

  @override
  int get hashCode => Object.hash(filePath, line, column);

  @override
  String toString() => '$filePath:$line:$column';
}

File: cs2dart_roslyn_worker/Models/SourceLocation.cs
namespace Cs2DartRoslynWorker.Models;

/// <summary>
/// A source location within a file (line and column are 1-based).
/// </summary>
public sealed class SourceLocation
{
    public string FilePath { get; set; } = string.Empty;
    public int Line { get; set; }
    public int Column { get; set; }
}
